apps/condo/domains/resident/tasks/discoverServiceConsumersCron.task.js

```diff
@@ -0,0 +1,92 @@
+const { loadListByChunks } = require('@condo/domains/common/utils/serverSchema')
+const { discoverServiceConsumers, ServiceConsumer } = require('@condo/domains/resident/utils/serverSchema')
+
+const DV_SENDER = { dv: 1, sender: { dv: 1, fingerprint: 'discoverServiceConsumersCronTask' } }
+const logger = getLogger('discoverServiceConsumersCronTask')
+const redisClient = getRedisClient('discoverServiceConsumersCronTask')
+const REDIS_KEY = 'discoverServiceConsumersLastDate'
+
+/**
+ * @returns {Promise<void>}
+ */
+async function discoverServiceConsumersCronTask () {
+    const lastDate = await redisClient.get(REDIS_KEY)
+    if (!lastDate) {
+        const message = `No last date in redis. Please set the ${REDIS_KEY} key with date (for example, "set ${REDIS_KEY} ${dayjs().toISOString()}")`
+        logger.warn({ message })
+        throw new Error(message)
+    }
+
+    const { keystone: context } = getSchemaCtx('BillingAccount')
+
+    if (conf.NODE_ENV === 'test') {
+        set(context, ['req', 'headers', 'feature-flags'], getIsFeatureFlagsEnabled() ? 'true' : 'false')
+    }
+
+    const billingAccountsIds = new Set()
+    let maxDate = dayjs(lastDate)
+
+    await loadListByChunks({
+        context,
+        list: BillingAccount,
+        chunkSize: 100,
+        where: {
+            createdAt_gt: dayjs(lastDate).toISOString(),
+            deletedAt: null,
+        },
+        sortBy: ['createdAt_ASC'],
+        /**
+         * @param {BillingAccount[]} chunk
+         * @returns {BillingAccount[]}
+         */
+        chunkProcessor: (chunk) => {
+            chunk.forEach((billingAccount) => {
+                billingAccountsIds.add(billingAccount.id)
+                const createdAt = dayjs(billingAccount.createdAt)
+                maxDate = createdAt.isAfter(maxDate) ? createdAt : maxDate
+            })
+
+            return []
+        },
+    })
+
+    /**
+     * @type {DiscoverServiceConsumersInput}
+     */
+    const data = {
+        ...DV_SENDER,
+        billingAccountsIds: Array.from(billingAccountsIds),
+    }
+
+    try {
+        const result = await discoverServiceConsumers(context, data)
+        logger.info({ message: 'discoverServiceConsumersCronTask done', result })
+
+        const [lastCreatedServiceConsumer] = await ServiceConsumer.getAll(context, { id_in: get(result, ['statistics', 'createdIds'], []) }, {
+            sortBy: ['createdAt_DESC'],
+            first: 1,
+        })
+
+        if (lastCreatedServiceConsumer) {
+            redisClient.set(REDIS_KEY, dayjs(lastCreatedServiceConsumer.createdAt).toISOString())
+        }
+    } catch (err) {
+        logger.error({ message: 'discoverServiceConsumersCronTask fail', err })
+    }
+}
+
+module.exports = {
+    REDIS_KEY,
+    discoverServiceConsumersCronTask: createCronTask('discoverServiceConsumersCronTask', '* * * * 13', discoverServiceConsumersCronTask, { priority: 10 }),
```

Comment:
Did you enter the correct time?

Here throws an error for such a value (https://crontab.guru/#*_*_*_*_13)

Reply:
OMG! Fixed :)